cpp/test/mesh/generation.cpp

```diff
@@ -0,0 +1,43 @@
+#include <catch2/catch_test_macros.hpp>
+
+#include <mpi.h>
+
+#include "dolfinx/mesh/generation.h"
+
+using namespace dolfinx;
+
+TEST_CASE("Generation", "create_interval")
+{
+    auto mesh = mesh::create_interval(MPI_COMM_SELF, 4, {0., 1.});
```

Comment:
We generally avoid `auto` unless the function makes obvious what the type is, e.g. `std::make_shared`, or the type name is long and complicated.